addons/website_sale_delivery_mondialrelay/models/models.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    is_mondialrelay = fields.Boolean(compute='_compute_is_mondialrelay')
+
+    def _compute_is_mondialrelay(self):
+        for p in self:
+            p.is_mondialrelay = p.ref and p.ref.startswith('MR#')
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    def _prepare_sale_order_values(self, partner, pricelist):
+        self.ensure_one()
+        values = super()._prepare_sale_order_values(partner, pricelist)
+        # never use Mondial Relay shpping address as default.
+        shipping_address = self.env['res.partner'].browse(values['partner_shipping_id'])
+        if shipping_address.id != values['partner_invoice_id']:
+            if shipping_address.is_mondialrelay:
+                values['partner_shipping_id'] = values['partner_billing_id']
+        return values
+
+
+class ProviderMondialRelay(models.Model):
+    _inherit = 'delivery.carrier'
+
+    is_mondialrelay = fields.Boolean(compute='_compute_is_mondialrelay')
+    mondialrelay_ens = fields.Char(string='Brand Code', default='BDTEST  ', groups="base.group_system")
+    mondialrelay_packagetype = fields.Char(default="24R", groups="base.group_system")  # Advanced
+
+    def _compute_is_mondialrelay(self):
+        for c in self:
+            c.is_mondialrelay = c.product_id.default_code == "MR"
+
+    def base_on_rule_get_tracking_link(self, picking):
+        if self.is_mondialrelay:
+            return 'https://www.mondialrelay.com/public/permanent/tracking.aspx?ens=%(brand)s&exp=%(track)s&language=%(lang)s' % {
+                'brand': picking.carrier_id.mondialrelay_ens,
+                'track': picking.carrier_tracking_ref,
+                'lang': (picking.partner_id.lang or 'fr').split('_')[0],
+            }
+        else:
```

Comment:
don't need the `else`

Reply:
was not wrong but already removed by a review from a colleague